extensions/amazon-sam/runtime/src/main/java/io/quarkus/amazon/sam/SamHandler.java

```diff
@@ -0,0 +1,192 @@
+
+    @Override
+    public void handleRequest(InputStream inputStream, OutputStream outputStream, Context context) throws IOException {
+        try {
+            AwsProxyRequest request = objectReader.readValue(inputStream);
+            AwsProxyResponse resp = proxy(request);
+
+            objectWriter.writeValue(outputStream, resp);
+        } catch (JsonParseException e) {
+            LOG.error("Error while parsing request object stream", e);
+            objectMapper.writeValue(outputStream, exceptionHandler.handle(e));
+        } catch (JsonMappingException e) {
+            LOG.error("Error while mapping object to RequestType class", e);
+            objectMapper.writeValue(outputStream, exceptionHandler.handle(e));
+        } finally {
+            outputStream.flush();
+            outputStream.close();
+        }
+    }
+
+    private AwsProxyResponse proxy(AwsProxyRequest request) {
+        try {
+            VirtualClientConnection connection = VirtualClientConnection.connect(VertxHttpRecorder.VIRTUAL_HTTP);
+
+            connection.sendMessage(buildRequest(request));
+            connection.sendMessage(buildContent(request));
+
+            return buildResponse(connection);
+        } catch (Exception e) {
+            LOG.error("Error while handling request", e);
+
+            return exceptionHandler.handle(e);
+        }
+    }
+
+    private AwsProxyResponse buildResponse(VirtualClientConnection connection) throws InterruptedException {
+        AwsProxyResponse response = null;
+        ByteArrayOutputStream baos = null;
+        for (;;) {
+            // todo should we timeout? have a timeout config?
+            //log.info("waiting for message");
+            Object msg = connection.queue().poll(100, TimeUnit.MILLISECONDS);
+            try {
+                if (msg == null)
+                    continue;
+
+                if (msg instanceof HttpResponse) {
+                    HttpResponse res = (HttpResponse) msg;
+                    response = new AwsProxyResponse(res.status().code());
+                    for (Map.Entry<String, String> entry : res.headers()) {
+                        response.addHeader(entry.getKey(), entry.getValue());
+                    }
+                }
+                if (msg instanceof HttpContent) {
+                    HttpContent content = (HttpContent) msg;
+                    if (baos == null) {
+                        // todo what is right size?
+                        baos = new ByteArrayOutputStream(500);
+                    }
+                    int readable = content.content().readableBytes();
+                    for (int i = 0; i < readable; i++) {
+                        baos.write(content.content().readByte());
+                    }
+                }
+                if (msg instanceof LastHttpContent) {
+                    response.setBody(new String(baos.toByteArray(), StandardCharsets.UTF_8));
```

Comment:
This does not handle non-text content.  See Stuarts implementation here:

https://github.com/quarkusio/quarkus/blob/c58a1c1eace83126de787b5003f11a619227500c/extensions/amazon-lambda-http/runtime/src/main/java/io/quarkus/amazon/lambda/http/runtime/AwsHttpHandler.java#L127

Reply:
To be honest this code should be used in the sam module, too, but it doesn't exist on master.  what gives?